### DISCUSSION TITLE
Change share box

### DISCUSSION THREAD

**Discussion by alimdi** [YOURLS/YOURLS#3445] [Plugins]
Hi,

I'm trying to make several changes on the sharebox with a plugin but not sure on how to do it :

- Change or remove copybox
- Change or remove sharebox (the middle)

What I've understood is that we can do all the things above but none seems to help achieving the desired changes.

- [ ] Shunt the whole sharebox with `yourls_add_filter( 'shunt_share_box', 'custom_shunt_share_box' );` making it to true value
- [ ] Change the data of share_box_data with `yourls_add_filter( 'share_box_data', 'custom_share_data' );`
- [ ] Add something before the copybox with `yourls_add_action( 'shareboxes_before', 'custom_sharebox_before' );`
- [ ] Add something on the middle just before the sharebox with `yourls_add_action( 'shareboxes_middle', 'custom_sharebox_middle' );`
- [ ] Add something after the sharebox with `yourls_add_action( 'shareboxes_after', 'custom_sharebox_after' );`

Any help appreciated.

**Comment by ozh** [comment]
Indeed, if you pass anything but false to `shunt_share_box`, say custom HTML, it will use this HTML instead of proceeding with the function ` yourls_share_box()`

Something like : 
```php
yourls_add_filter( 'shunt_share_box', 'custom_shunt_share_box' );

function custom_shunt_share_box() {
    // some tests depending on context ?

    // if you want to return your custom share box:
    $html = ....
    return $html;

    // if you want the default share box
    return false;
}
```

There are numerous plugins about the share box -- see https://github.com/YOURLS/awesome-yourls and search for "Share", maybe this will help ?
